src/services/filesync/filesync.ts

```diff
@@ -152,16 +161,33 @@ export class FileSync {
     // try to load the .gadget/sync.json file
     const state = await fs
       .readJson(path.join(dir, ".gadget/sync.json"))
-      .then((json) =>
-        z
+      .then((json) => {
+        return z
           .object({
             app: z.string(),
-            filesVersion: z.string(),
-            mtime: z.number(),
+            filesVersion: z
+              .string()
+              .nullish()
+              .transform((x) => x ?? "0"), // Legacy support for old sync.json files
+            currentEnvironment: z
+              .string()
+              .nullish()
+              .transform((x) => x ?? "development"), // New support for multi environment, but old sync.json files don't have this
+            environments: z
+              .record(z.object({ filesVersion: z.string() }))
+              .nullish()
+              .transform((value, _ctx) => value ?? { development: { filesVersion: "0" } }),
           })
-          .parse(json),
-      )
-      .catch(noop);
+          .parse(json);
+      })
+      .catch((e) => ctx.log.warn("failed to load .gadget/sync.json", { error: e }));
+
+    // Basically a hack for reverse compatibility with old sync.json files
+    const devEnvironmentState = state?.environments["development"];
```

Comment:
I'm a little conflicted on this, the only reason I can think of to do this is to support people updating to this version of GGT and backdown?

Reply:
@scott-rc made a good point last week, sometimes people will switch between using npx to run `ggt` and in that case it'll easily switch between two versions. So I'm cool keeping this.